src/main/java/org/yourteamname/agents/MessageQueueAgent.java

```diff
@@ -0,0 +1,203 @@
+package src.main.java.org.yourteamname.agents;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import jade.core.Agent;
+import jade.core.behaviours.CyclicBehaviour;
+import jade.core.behaviours.TickerBehaviour;
+import jade.domain.DFService;
+import jade.domain.FIPAException;
+import jade.domain.FIPAAgentManagement.DFAgentDescription;
+import jade.domain.FIPAAgentManagement.ServiceDescription;
+import jade.lang.acl.ACLMessage;
+import jade.lang.acl.MessageTemplate;
+import jade.lang.acl.UnreadableException;
+
+public class MessageQueueAgent extends Agent {
+	private static final long serialVersionUID = -5310054528477305012L;
+	private boolean gui = false;
+
+	private List<AgentQueueMap> agentQueueMaps;
+
+	// Put agent initializations here
+	@SuppressWarnings("serial")
+	protected void setup() {
+		Object[] args = getArguments();
+		if (args != null && args.length > 0) {
+			gui = (boolean) args[0];
+		}
+
+		agentQueueMaps = new EventPrintList();
+
+		// Register service in the yellow pages
+		DFAgentDescription dfd = new DFAgentDescription();
+		dfd.setName(getAID());
+		ServiceDescription sd = new ServiceDescription();
+		sd.setType("message-queue");
```

Comment:
The services offered should reflect those of the loading bay agent

Reply:
As you can surely tell the message queue agent is the target of this pull request. This pull request does not contain any of the code we use for the packaging agent or the loading bay. Those agents are currently worked on by the rest of my team and I guess they will create a pull request as soon as they are done with it and find the time to upload the code.
@HBRS-MAAS/team-trk

Reply:
@argenos Since this pull request apparently does not deal with the loadingBay agent, are we still supposed to review it?

Thanks!
@HBRS-MAAS/commitment_issues

Reply:
@HBRS-MAAS/team-trk unfortunately, the original deadline for the agents was yesterday evening, and since there was no pull request for that, I assumed this was all there is to grade. 

@HBRS-MAAS/commitment_issues No, please disregard this pull request from this week's assignment.